plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1788,6 +1797,26 @@ else if (partitionUpdate.getUpdateMode() == NEW || partitionUpdate.getUpdateMode
                         .collect(toImmutableList())));
     }
 
+    private void removeNonCurrentQueryFiles(ConnectorSession session, Optional<HiveBucketProperty> hiveBucketProperty, Path partitionPath)
+    {
+        try {
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(new HdfsContext(session), partitionPath);
+            RemoteIterator<LocatedFileStatus> iterator = fileSystem.listFiles(partitionPath, false);
+            while (iterator.hasNext()) {
+                Path file = iterator.next().getPath();
+                String fileName = file.getName();
+                String queryId = session.getQueryId();
+                if ((hiveBucketProperty.isEmpty() && !fileName.startsWith(queryId))
+                        || hiveBucketProperty.isPresent() && !fileName.endsWith(queryId)) {
```

Comment:
can we extract this logic to some helper util method? Maybe it exists already?

Reply:
Extracted to HiveWriteUtils